src/Stamp.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { space } from 'styled-system'
+import theme from './theme'
+
+const type = props => {
+  const StampColors = {
+    gray: {
+      backgroundColor: props.theme.colors.lightGray,
+      color: props.theme.colors.gray
+    },
+    blue: {
+      backgroundColor: props.theme.colors.lightGray,
+      color: props.theme.colors.blue
+    },
+    green: {
+      backgroundColor: props.theme.colors.lightGray,
+      color: props.theme.colors.green
+    },
+    red: {
+      backgroundColor: props.theme.colors.lightGray,
+      color: props.theme.colors.red
+    },
+    orange: {
+      backgroundColor: props.theme.colors.lightGray,
+      color: props.theme.colors.orange
+    },
+    purple: {
+      backgroundColor: props.theme.colors.lightGray,
+      color: props.theme.colors.purple
+    }
+  }
+  return StampColors[props.color]
+}
+
+const Stamp = styled.div`
+  display: inline-flex;
+  align-items: center;
+  vertical-align: top;
+  min-height: 24px;
+  font-size: ${props => props.theme.fontSizes[0]}px;
```

Comment:
NB: I'd say it might be better to handle font-size with the `fontSize` utility and a default prop for code consistency with the other components